src/DependencyInjection/SonataPageExtension.php

```diff
@@ -111,7 +113,18 @@ public function load(array $configs, ContainerBuilder $container)
             ->replaceArgument(1, $config['ignore_route_patterns'])
             ->replaceArgument(2, $config['ignore_uri_patterns']);
 
-        $this->registerDoctrineMapping($config);
+        if (isset($bundles['SonataDoctrineBundle'])) {
+            $this->registerSonataDoctrineMapping($config);
+        } else {
+            // NEXT MAJOR: Throw error when not registering SonataDoctrineBundle
+            @trigger_error(
+                'Using SonataEasyExtendsBundle is deprecated since sonata-project/timeline-bundle 3.x. Please register SonataDoctrineBundle as a bundle instead.',
+                E_USER_DEPRECATED
+            );
+
+            $this->registerDoctrineMapping($config);
```

Comment:
Should we deprecate the `registerDoctrineMapping` method then and add a NEXT_MAJOR comment about it ?

Reply:
To avoid changing it on all PRs at once.. WDYT using this?

```
    // NEXT_MAJOR: Remove this method
    public function registerDoctrineMapping(array $config): void
    {
        @trigger_error(
            'Using this method is deprecated since sonata-project/user-bundle 4.x. You should instead register SonataDoctrineBundle and use `registerSonataDoctrineMapping()`',
            E_USER_DEPRECATED
        );
```

Reply:
Seems ok to me but if we add a deprecation in the method, I don't think we also need the deprecation right before the method call (in the else). One will be enough, don't you think ?

Reply:
Yes, we should only trigger one deprecation. Wdyt if I move the deprecation from outside the method instead of changing the message?

Reply:
Seems perfect

Reply:
wdyt about the rest of the PR? it would be nice to make all the changes at once.

Reply:
I don't really know how all this registration works, but I had nothing more to say about the code.

I added some comment about the doc.